.github/actions/csv-vulnerability-filter/src/main.ts

```diff
@@ -0,0 +1,193 @@
+      const severityLevels = ['CRITICAL', 'HIGH', 'MEDIUM', 'LOW', 'INFORMATIONAL'];
+      const severityIndex = severityLevels.indexOf(minSeverity.toUpperCase());
+      if (severityIndex === -1) {
+        core.warning(`Invalid severity level "${minSeverity}". No filtering applied.`);
+      } else {
+        const prefilterLength = filtered.length;
+        filtered = filtered.filter(row => {
+          const severity = row['Severity']?.toUpperCase();
+          return severityLevels.indexOf(severity) <= severityIndex;
+        });
+        const postfilterLength = filtered.length;
+        const removedCount = prefilterLength - postfilterLength;
+        if (removedCount > 0) {
+          core.info(`Removed ${removedCount} findings below severity level "${minSeverity}".`);
+        }
+      }
+    }
+
+    // --- Generate Output CSV Strings ---
+    const filteredCsvString = generateCsvString(filtered);
+    const removedCsvString = generateCsvString(removed);
+
+    // --- Write Output CSV Files ---
+    const filteredCsvOutputPath = writeOutputCsv(filteredCsvString, 'filtered_results');
+    const removedCsvOutputPath = writeOutputCsv(removedCsvString, 'removed_results');
+
+    // --- Generate Output Markdown ---
+    core.info('Generating Markdown reports...');
+    let filteredMarkdown = generateMarkdown(filtered, primaryData);
+    let removedMarkdown = generateMarkdown(removed);
+
+    if (comparisonSkipped) {
+      removedMarkdown = `## Vulnerability Findings (Removed by Comparison)\n\nBaseline comparison was skipped (no comparison file provided or file was unparsable).\n`;
+    } else if (removed.length === 0) {
+      removedMarkdown = `## Vulnerability Findings (Removed by Comparison)\n\nNo findings were removed based on the comparison file.\n`;
+    }
+    if (comparisonSkipped) {
+      filteredMarkdown = filteredMarkdown.replace('## Vulnerability Findings', '## Vulnerability Findings (Primary Scan)');
+    } else {
+      filteredMarkdown = filteredMarkdown.replace('## Vulnerability Findings', '## Vulnerability Findings (Filtered)');
+    }
+
+    // --- Add Markdown to GitHub Job Summary ---
+    if (process.env.GITHUB_STEP_SUMMARY) {
+      core.info('Appending filtered results to GitHub Job Summary...');
+      core.summary.addRaw(filteredMarkdown);
+      await core.summary.write();
+      core.info('Successfully wrote to GitHub Job Summary.');
+    } else {
+      core.info('GITHUB_STEP_SUMMARY environment variable not set. Skipping job summary update.');
+    }
+
+    // --- Write Output Markdown Files ---
+    const filteredMarkdownOutputPath = writeOutputMarkdown(filteredMarkdown, 'filtered_results');
+    const removedMarkdownOutputPath = writeOutputMarkdown(removedMarkdown, 'removed_results');
+
+    // --- Check Thresholds ---
+    core.info('Calculating severity counts for threshold check...');
+    const filteredCounts = calculateSeverityCounts(filtered);
+    const thresholdExceeded = 1;
```

Comment:
Should this call `checkCountThresholds` instead of hardcoding it to 1?

Reply:
Yep!